services/QuillLMS/app/services/classlink_integration/auth_credential_saver.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module ClasslinkIntegration
+  class AuthCredentialSaver < ApplicationService
+    JWT_PASSWORD = nil
+    JWT_SHOULD_VALIDATE = false
+    attr_reader :code
+
+    def initialize(code:)
+      @code = code
+    end
+
+    def run
+      fetch_credentials
+        .then { |cs| AuthCredentialSaver.format_credentials(cs) }
+        .then { |fcs| AuthCredentialSaver.generate_auth_credential(**fcs) }
+    end
+
+    def fetch_credentials
+      uri = URI.parse(::Auth::Classlink::OAUTH_TOKEN_ENDPOINT)
+      form_data = {
+        client_id: ::Auth::Classlink::CLIENT_ID,
+        client_secret: ::Auth::Classlink::CLIENT_SECRET,
+        code:
+      }
+
+      Net::HTTP.post_form(uri, form_data)
+    end
+
+    def self.generate_auth_credential(access_token:, user_sourced_id:)
+      account = ClasslinkAccount.find_by(external_id: user_sourced_id)
```

Comment:
Will `account` here ever be nil?  If so, we probably need error handling to protect against `account.user_id`

Reply:
The sign[in up] behavior of users pre-district sync is not fully spec'd out yet, although Nattalie and I have stubbed specs in the project, so we don't overlook this issue. Thus, my preference is to fail loudly, until the full spec is implemented.